app/src/main/java/org/mozilla/fenix/settings/creditcards/CreditCardEditorFragment.kt

```diff
@@ -59,10 +62,20 @@ class CreditCardEditorFragment : SecureFragment(R.layout.fragment_credit_card_ed
 
         creditCardEditorState =
             args.creditCard?.toCreditCardEditorState() ?: getInitialCreditCardEditorState()
-        creditCardEditorView = CreditCardEditorView(view, interactor)
+
+        creditCardEditorView = CreditCardEditorView(view, interactor, creditCardValidator)
         creditCardEditorView.bind(creditCardEditorState)
     }
 
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        val storage = requireContext().components.core.autofillStorage
+        lifecycleScope.launch(Dispatchers.IO) {
+            val list = storage.getAllCreditCards()
```

Comment:
I think we want to call this on demand when we do validate the credit card. Otherwise, it's possible that we want to the edge case where someone was just hovering on the CreditCardEditorFragment and meanwhile in sync, there's a new credit card that has been added, but we won't be able to identify that as a duplicate since we cache the list of credits. So, my point here is to avoid caching the lists of credit cards if possible since we only check if it's duplicate when the save button is clicked.

Reply:
While I agree there might be some edge cases, the current implementation helps to get a faster validation response that is useful in all cases, not just in an edge case.